packages/contracts-bedrock/src/cannon/PreimageOracle.sol

```diff
@@ -549,9 +550,8 @@ contract PreimageOracle is ISemver {
         LPPMetaData metaData = proposalMetadata[msg.sender][_uuid];
         uint256 blocksProcessed = metaData.blocksProcessed();
 
-        // The caller of `addLeavesLPP` must be an EOA.
-        // Note: This check may break if EIPs like EIP-3074 are introduced. We may query the data in the logs if this
-        // is the case.
+        // The caller of `addLeavesLPP` must be an EOA, so that the call inputs are always available in block bodies.
+        // We do NOT use the EOA library here so that we can guarantee that the call inputs are available.
         if (msg.sender != tx.origin) revert NotEOA();
```

Comment:
I wonder if we should delete both of these EOA checks in PreimageOracle? Per @ajsutton 

> It has already been handled. That check was originally in place because the challenger had to load the post data for large preimage proposals from the tx call data. But when 7702 or an earlier variant was going to ship in the last hard fork or the one before it right around the time we launched fault proofs we actually switched to publishing the added data in an event and the challenger loads it from there. We kept the eoa check out of an abundance of caution but it’s not actually needed.

Though I don't see an event needed in `initLPP` so maybe that one does still need the check?

Reply:
Strongly prefer to keep the existing code to keep diff down